File: api/src/EventList.ts
import { Evenement } from './Evenement'

export class EventList {
  private readonly eventList: Evenement[] = []

  addEvent(event: Evenement) {
    if(this.eventList.length == 0) {
      this.eventList.push(event);
    }

    else{
      for(let i = 0; i < this.eventList.length; i++) {
        if(this.eventList[i].title.localeCompare(event.id) == 0) {
          return;
        }
      }
      this.eventList.push(event);
    }
  }

  getEvent(id: string): Evenement | undefined {
    let chosenEvent;

    if(this.eventList.length == 0) {
      return;
    }

    else
    {
      for(let i = 0; i < this.eventList.length; i++)
      {
        if(this.eventList[i].id.localeCompare(id) == 0)
          chosenEvent = this.eventList[i];
        return chosenEvent;
      }
    }
  }

  getAllEvents(): Evenement[] | undefined {
    return this.eventList.sort((bookA, bookB) => bookA.date_start.localeCompare(bookB.date_start));
  }

  getTotalNumberOfEvents() : number {
    return this.eventList.length;
  }
}

File: android/app/src/main/java/com/ismin/projectapp/EventList.kt
/**
 * @authors : Valentin POLLART - Fatima-Zohra NAZIH
 * @title : Events in Paris
 */

package com.ismin.projectapp

class EventList {
    private val storage: ArrayList<Event> = ArrayList()

    fun addEvent(newEvent: Event) {
        if(this.getEventByID(newEvent.id) == null) { this.storage.add(newEvent) }
    }

    fun getEvent(title: String): Event? {
        return this.storage.find{it.title == title}
    }

    fun getEventByID(id: String): Event? {
        return this.storage.find { it.id === id }
    }

    fun getAllEventsSortedByTitle(): ArrayList<Event> {
        return ArrayList(this.storage.sortedBy { it.title })
    }

    fun getAllEventsSortedByDate(): ArrayList<Event> {
        return ArrayList(this.storage.sortedBy { it.date_start })
    }

    fun getTotalNumberOfEvents(): Int {
        return this.storage.size
    }

    fun removeEvent(event: Event) {
        this.storage.remove(event)
    }
}